lib/statesman/config.rb

```diff
@@ -5,14 +5,19 @@
 
 module Statesman
   class Config
-    attr_reader :adapter_class
+    attr_reader :adapter_class, :gaplock_protection_enabled
 
     def initialize(block = nil)
+      @gaplock_protection_enabled = false
```

Comment:
We've mixed calling this gaplock_protection and mysql_gaplock_protection. Can we change things so we have one uniform reference, preferably `mysql_gaplock_protection` in all places?

Reply:
I think we should set this parameter to be `true` when the adapter looks like mysql. Otherwise all the mysql statesman users end up with a crucial performance optimisation not enabled, which sucks.

Wdyt?

Reply:
My thoughts on this were that we don't want people with postgres enabling this, so the API from the config perspective should make that clear. We do something similar in `storage_adapter`. No strong opinions though, did that make sense?

I agree we should enable this by default, I can't remember why I didn't.